### DISCUSSION TITLE
Disable camera movements when replaying a recorded game session

### DISCUSSION THREAD

**Discussion by deurklink** [OpenRCT2/OpenRCT2#19596] [Feature requests] [answered]
I want to use the replay functionality to record time lapses for the parks I build, so I get a more clean video without all scenery and ride windows popping up every time, and more importantly, without me moving and rotating the screen every time while I am building.

It works mostly fine, I can record and replay a session just fine, the only annoyance I have now is that the camera moves to the location of the action every time an action is replayed.. I think it would be really nice if I could disable this, so I have full control of the camera while I record my time lapse :)

**Comment by deurklink** [comment]
So basically, it would be really nice if I could somehow disable the following code in ReplayManager.cpp
```
                // Focus camera on event.
                if (isPositionValid && !result.Position.IsNull())
                {
                    auto* mainWindow = WindowGetMain();
                    if (mainWindow != nullptr)
                        WindowScrollToLocation(*mainWindow, result.Position);
                }
```

**Comment by Broxzier** [comment]
The replay was mainly created to validate the game's integrity, so no accidental changes get merged. It's interesting to see the other use of the feature, though!

Another way to accomplish a time-lapse effect is by having a server and client running. Build one one of them, and record the other window. A downside is that you need to run two instances of the game, and that you cannot make adjustments to the object selection.

**Reply by deurklink** [reply]
Having to run a server would indeed be a bit annoying for the reasons you mentioned, I'm also not sure if all my plugins would work correctly without causing desyncs. As @ZehMatt mentioned, the most straightforward solution would probably be to disable these lines of code that move the camera with an option in the config file.

**Reply by ZehMatt** [reply]
I will think a bit about this, having it behind the config is not necessarily the best choice also, perhaps an optional argument might be the better choice. I also want to look into adding the replay APIs to the scripting system, creating UI with plugins is a lot easier.

**Reply by deurklink** [reply]
If we do this, we probably also want to make it so that we can toggle the text "Replaying..." being shown on screen. While it is handy in other situations, for a recording this is probably not what you want to have flashing in the screen.

**Reply by fidwell** [reply]
This is a very intriguing idea! I agree with ZehMatt; since this seems like a very advanced thing only a few people will probably ever take advantage of (for now anyway), having an optional startup argument makes the most sense to me. We could have one flag that toggles having either the original "regression check" mode, or having "clean video" mode for recording playback.

**Comment by fidwell** [comment, marked as answer]
Resolved with #22303
